### ISSUE TITLE
Update Unity to 2019 LTS

### ISSUE THREAD

**Issue by cosmiccoincidence** [RE-SS3D/SS3D#655]
### Summary

We're still on like 2019.3.12.... wth... time to update!

**Comment by stilnat** [comment]
I suggest updating to version 2019.4.28f1 or version 2020.3.10f1. They are both LTS versions of Unity. Furthermore, I noticed that Visual Studio sync way better with Unity when I update the project to version 2019.4.28f1, it's pretty much out of the box from my tests. With the actual version, I had quite a fun time trying to sync Visual studio and had to give up eventually. That's one of the perks I see from upgrading into this version, and I'll try with 2020.3.10f1 too.

**Comment by cosmiccoincidence** [comment]
Many of us want to update to *at least* 2020 due to the integrated (free) dark mode.

**Comment by stilnat** [comment]
Ok so here's my experimentations with Unity 2020 so far. I downloaded Unity 2020.3.11 the LTS version so far. 

 I downloaded a clean clone of the project, and upgraded it to 2020.3.11. First thing after the upgrade, Unity asks for urp material upgrade. I accepted as it's mandatory. I tried launching the main station scene, and run into a few errors : 

`InvalidProgramException: Invalid IL code in SS3D.Content.Systems.Substances.SubstanceDisplay:UpdateDisplay (): IL_000f: call      0x0a000019.`
This one appears right after loading the map.

`InvalidProgramException: Invalid IL code in SS3D.Engine.Server.Round.RoundManager:UpdateClock (int): IL_000f: call      0x0a000019.`
This one when starting the round.

They're both network related errors as shown below, Mirror is involved (I'm showing only for the second one, but the first one is similar) : 

```
SS3D.Engine.Server.Round.RoundManager+<TickWarmup>d__38.MoveNext () (at Assets/Engine/Server/Round/RoundManager.cs:167)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <be28439f7b014fc38435d198c0640ff7>:0)
UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
SS3D.Engine.Server.Round.RoundManager:StartWarmup() (at Assets/Engine/Server/Round/RoundManager.cs:53)
SS3D.SceneLoaderManager:HandleRoundButton() (at Assets/Engine/Server/Mirror/SceneLoaderManager.cs:247)
SS3D.SceneLoaderManager:<Awake>b__15_1() (at Assets/Engine/Server/Mirror/SceneLoaderManager.cs:63)
UnityEngine.EventSystems.EventSystem:Update() (at C:/Program Files/Unity/Hub/Editor/2020.3.10f1/Editor/Data/Resources/PackageManager/BuiltInPackages/com.unity.ugui/Runtime/EventSystem/EventSystem.cs:385)
```

That's pretty obscure to me as of what's going on there. It just reminded me of the [link posted by](https://mirror-networking.gitbook.io/docs/general/recommended-unity-version) @Ryan089 warning about Mirror stability in 2020 versions of Unity . Fixing those errors is beyond my understanding for now, so maybe someone else can take a look into it ? 

I also found [this guide to update from 2019 to 2020 ](https://docs.unity3d.com/2020.1/Documentation/Manual/UpgradeGuide20201.html), which could be useful to someone with a better overall understanding of the project. 
For example, do we use the HLAPI package somewhere ? 

That's where I'm at with the 2020 version attempt to update.
